modules/sdk-coin-avaxp/src/lib/transaction.ts

```diff
@@ -146,7 +155,41 @@ export class Transaction extends BaseTransaction {
    */
   get signablePayload(): Buffer {
     const txbuff = this._avaxpTransaction.toBuffer();
-    return createHash.default('sha256').update(txbuff).digest();
+    return utils.sha256(txbuff);
+  }
+
+  get id(): string {
+    return utils.cb58Encode(BufferAvax.from(utils.sha256(utils.cb58Decode(this.toBroadcastFormat()))));
+  }
+
+  get fromAddresses(): string[] {
+    return this._fromAddresses.map((a) => utils.addressToString(this._network.hrp, this._network.alias, a));
+  }
+  /**
+   * Get the list of outputs. Amounts are expressed in absolute value.
+   */
+  get outputs(): Entry[] {
+    if (this.type === TransactionType.addValidator) {
+      const addValidatorTx = this.avaxPTransaction as AddValidatorTx;
+      return [
+        {
+          address: addValidatorTx.getNodeIDString(),
+          value: addValidatorTx.getStakeAmount().toString(),
+        },
+      ];
+    }
+    // general support any transaction type, but it's scoped yet
+    return this.avaxPTransaction.getOuts().map((output) => {
+      const amountOutput = output.getOutput() as any as AmountOutput;
+      const address = amountOutput
+        .getAddresses()
```

Comment:
nit: an output can send to multiple addresses? Is it appropriate to join these with a `, ` or would creating a separate entry per output address be better?

Reply:
It's one entry. Multisig account uses the addresses of its signer and doesn't have a multisigner address representations.However we still need to review this.